### DISCUSSION TITLE
File Upload and general validation

### DISCUSSION THREAD

**Discussion by jasonbdt** [inertiajs/inertia#514] [Help]
Hello again,

today I got a new problem according to file uploads with inertiajs and laravel. I've implemented a validation rule that checks if the maximum file size is exceeded or not. When I use

```php
  return $validator->errors();
```

I get this response:

```json
{"file":["The file failed to upload."],"file_upload":["Unexpected error occured"]}
```

But the $page.props.errors Object is always empty. 
Before I updated all dependencies (laravel / inertia) to it's latest version it worked.

#Edit: The only form where validation works is on login and forgot password page.

**package.json**
```
    "devDependencies": {
        "@inertiajs/inertia": "^0.8.0",
        "@inertiajs/inertia-vue": "^0.5.0",
        "@tailwindcss/aspect-ratio": "^0.2.0",
        "@tailwindcss/forms": "^0.2.1",
        "@tailwindcss/postcss7-compat": "^2.0.2",
        "@tailwindcss/typography": "^0.3.1",
        "autoprefixer": "^9.8.6",
        "axios": "^0.19",
        "cross-env": "^7.0",
        "laravel-mix": "^5.0.1",
        "lodash": "^4.17.19",
        "moment": "^2.26.0",
        "portal-vue": "^2.1.7",
        "postcss": "^7.0.35",
        "postcss-import": "^12.0.1",
        "resolve-url-loader": "^3.1.0",
        "tailwindcss": "npm:@tailwindcss/postcss7-compat@^2.0.2",
        "vue": "^2.5.17",
        "vue-slim-tabs": "^0.4.0",
        "vue-template-compiler": "^2.6.10"
    },
    "dependencies": {
        "vue-material-design-icons": "^4.11.0"
    }
```

**composer.json**
```
    "require": {
        "php": "^7.4|^8.0",
        "fideloper/proxy": "^4.4",
        "fruitcake/laravel-cors": "^2.0",
        "guzzlehttp/guzzle": "^7.0.1",
        "inertiajs/inertia-laravel": "^0.3",
        "laravel/framework": "^8.12",
        "laravel/jetstream": "^2.0",
        "laravel/sanctum": "^2.6",
        "laravel/tinker": "^2.5",
        "tightenco/ziggy": "^0.9.4"
    },
    "require-dev": {
        "facade/ignition": "^2.5",
        "fakerphp/faker": "^1.9.1",
        "mockery/mockery": "^1.4.2",
        "nunomaduro/collision": "^5.0",
        "phpunit/phpunit": "^9.3.3"
    },
```

Can someone help and lead me to the right way?

Regards,
Jason

**Comment by reinink** [comment]
Can you show your controller code?

**Comment by reinink** [comment]
Also, it might be worth reviewing how we do this in Ping CRM:

- [Controller](https://github.com/inertiajs/pingcrm/blob/master/app/Http/Controllers/UsersController.php#L49)
- [Page component](https://github.com/inertiajs/pingcrm/blob/master/resources/js/Pages/Users/Create.vue#L18)

**Reply by jasonbdt** [reply]
Okay, this is interesting... if I try to use these two it don't works (page prop `errors` not updated)
```php
Validator::make($request->all(), [ 'foo' => ['required', 'max:255'] ]);

$request->validate([ 'foo' => ['required', 'max:255'] ])
```

But doing it like shown in your examples it works:

```php
use Illuminate\Support\Facades\Request;

Request::validate([ 'foo' => ['required', 'max:255'] ]);
```

Is it maybe important which Request-Class is imported? Laravel ships with
- Illuminate\Http\Request
- Illuminate\Support\Facades\Request

any maybe much more.

**Comment by jasonbdt** [comment]
For me it feels like that the updating of `errors` prop is missing in the laravel adapter when using `Validator::make()`.
Or is there any reason behind that?

**Comment by reinink** [comment]
So, I just tried both the `Validator::make()` and `$request->validate()` approaches, and they both worked for me.

Keep in mind that when you use `Validator::make()`, you need to call `validate()` on it to actually trigger the validation:

```php
use Illuminate\Support\Facades\Validator;

Validator::make($request->all(), [ 'foo' => ['required', 'max:255'] ])->validate();
```
